### PR TITLE
Split image repositories into registry and repository

### PR DESCRIPTION
**Pull request by fhielpos**
## Related issues
- Close https://github.com/aquasecurity/trivy-operator/issues/1135

Remove this section if you don't have related PRs.

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by chen-keinan** [comment]
@fhielpos you'll have to run `make generate-all` from project folder and commit changes for verify code issue

**Comment by fhielpos** [comment]
Thanks @chen-keinan, had to also nest the Trivy server values since we already have a `registry` value there. Can you let the CI run again?

File: deploy/helm/README.md [hunk 1]
```diff
@@ -110,9 +110,11 @@ The values of the `OPERATOR_NAMESPACE` and `OPERATOR_TARGET_NAMESPACES` determin
...
+| `trivy.image.tag`                              | `0.36.0`                           | Version of the Trivy image                                                                                                                                                                                                                                                                                                                                                                       |
+| `trivy.image.imagePullSecret`                  | N/A                                | imagePullSecret is the secret name to be used when pulling trivy image from private registries example: `reg-secret`. It is the user responsibility to create the secret for the private registry in `trivy-operator` namespace.                                                                                                                                                                 |
```
Comment (chen-keinan) [inline deploy/helm/README.md:117]: these doc represent trivy config map and you changes impact only help setting the `cm` is not changing correct ?

**Comment by chen-keinan** [comment]
@fhielpos thank you for the contribution , I have added another comment

File: deploy/helm/README.md (hunk 1 above)
Comment (fhielpos) [reply deploy/helm/README.md:117]: Yes, only the helm values are being changed here. The ConfigMap itself would still be the same.

Comment (fhielpos) [reply deploy/helm/README.md:117]: Oh, yes! Now I got your point. Will revert that one 😄

**Comment by fhielpos** [comment]
Thanks for your comments @chen-keinan , all done 😄

**Comment by chen-keinan** [comment]
lgtm , thank you for the contribution 🚀

**Review by chen-keinan — APPROVED** [review]

**Pull request merged**